ddpui/api/charts_api.py

```
@@ -0,0 +1,1010 @@
+
+        # Validate that all features have the specified properties_key
+        for i, feature in enumerate(features):
+            properties = feature.get("properties", {})
+            if payload.properties_key not in properties:
+                raise HttpError(
+                    400, f"Feature {i+1} missing required property: {payload.properties_key}"
+                )
+
+    except Exception as e:
+        logger.error(f"GeoJSON validation error: {str(e)}")
+        raise HttpError(400, f"Invalid GeoJSON: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)